File: practice/mathematics/fundamentals/halloween_party/HalloweenParty.scala
import scala.io.Source

object HalloweenParty extends App {
  val lines = Source.stdin.getLines()
  val slices = lines.drop(1).map(_.toLong)
  val result = slices.map(pieces)
  println(result.mkString("\n"))

  def pieces(k: Long): Long = {
    val horizontal = k / 2
    val vertical = k - horizontal
    vertical * horizontal
  }
}

File: practice/mathematics/fundamentals/halloween_party/HalloweenParty.js
'use strict';

const processData = input => {
    let lines = input.split("\n");
    let slices = lines.map(i => parseInt(i));
    const pieces = k => {
        let horizontal = parseInt(k / 2);
        let vertical = k - horizontal;
        return vertical * horizontal;
    };
    for(let i = 1; i < slices.length; i++) {
        console.log(pieces(slices[i]));
    }
};

process.stdin.resume();
process.stdin.setEncoding("ascii");

var _input = "";
process.stdin.on("data", input => _input += input);
process.stdin.on("end", () => processData(_input));